operator/duties/scheduler.go

```diff
@@ -383,11 +371,12 @@ func (s *Scheduler) ExecuteGenesisDuties(logger *zap.Logger, duties []*genesissp
 		if slotDelay >= 100*time.Millisecond {
 			logger.Debug("⚠️ late duty execution", zap.Int64("slot_delay", slotDelay.Milliseconds()))
 		}
-		slotDelayHistogram.Observe(float64(slotDelay.Milliseconds()))
+		slotDelayHistogram.Record(context.TODO(), slotDelay.Seconds())
```

Comment:
we could probably use use the context passed in `SchedulerOptions` (need to save it in the `Scheduler` struct).

Reply:
fixed. Added a `ctx` field to the Scheduler object that is passed through ctor as part of `SchedulerOptions` and switched to use it instead of `context.TODO()`.